M2/Macaulay2/packages/ResolutionsOfStanleyReisnerRings.m2

```diff
@@ -0,0 +1,589 @@
+
+newPackage(
+    "ResolutionsOfStanleyReisnerRings",
+    Version => "0.1",
+    Date => "July 15, 2020",
+    	Authors => {{Name => "Ashleigh Adams", Email => "adams869@umn.edu", HomePage => "http://www.ashleigh-adams.com"}},
+    Headline => "Comparing resolutions of Stanley-Reisner rings and computing various systems of parameters",
+    PackageExports => {"SimplicialComplexes", "Posets", "SimplicialDecomposability"},
+    DebuggingMode => false  
+    )
+
+-------------------
+-- Exports
+------------------- 
+export {"colorfulSOP","universalSOP","colorfulPresentation","universalPresentation","barycentricSubdivision","sumBetti","equalBettiTally"}
+exportMutable("Colors")
+
+-------------------
+-- Exported Code
+-------------------
+
+-----------------------------------------
+--Barycentric Subdivision-
+-----------------------------------------
+barycentricSubdivision = method(TypicalValue => SimplicialComplex)
+barycentricSubdivision (SimplicialComplex) := D -> (
+   return orderComplex(facePoset(D),CoefficientRing=>coefficientRing(ring(D)));
+);
+
+barycentricSubdivision (List) := F -> (
+   D := simplicialComplex(F);
+   return orderComplex(facePoset(D),CoefficientRing=>coefficientRing(ring(D)));
+);
+
+---------------------------------------------------
+--Non-exported code for Colorful Version
+---------------------------------------------------
+
+-- Used in balancedColoring function, it returns the color of a vertex
+vertexColor = (a,Colors) -> (
+   for i from 0 to #Colors-1 do ( if member(a,Colors#i) then return i; );
+);
+
+-----------------------------------------
+--Colorful System of Parameters  
+-----------------------------------------
+
+colorfulSOP = method(TypicalValue => List)
+colorfulSOP (SimplicialComplex,ZZ) := (B,n) -> (
+   R      := ring(B);
+   Colors := new MutableList from {};
+   for i from 0 to dim(B) do ( Colors = append(Colors,{}); );
+   for i from 1 to n do ( Colors#0 = append(Colors#0,R_i); );
+   ColorsNotAllowed := {0};
+   AlreadyColored   := splice{1..#Colors#0};
+   NotColored       := splice{#Colors#0+1..#faces(0,B,useFaceClass=>true)};
+   
+   --Gives a proper coloring to the graph of B
+   for i from 1 to #Colors#0 do (
+      for c in NotColored do (
+         if isFaceOf(face{R_i,R_c},B) then (
+            for a in AlreadyColored do (
+               if isFaceOf(face{R_c,R_a},B) then (
+                  ColorOfVertA := vertexColor(R_a,Colors);
+                     if not member(ColorOfVertA,ColorsNotAllowed) then (
+                        ColorsNotAllowed = append(ColorsNotAllowed,ColorOfVertA);
+                     );
+               );
+             );
+         );
+         m  := max(ColorsNotAllowed) + 1;
+         Colors#m       = append(Colors#m,R_c);
+         AlreadyColored = append(AlreadyColored,c);
+         NotColored     = delete(c,NotColored);
+      );
+   );
+   C := {};
+   for j from 0 to dim(B) do (
+       C = append(C,(sum for i in Colors#j list i));
```

Comment:
This is a slow way to make a list.  Can you just use `apply` on the list?

Reply:
Fantastic. So much better. Thanks!